### DISCUSSION TITLE
Trynig to create Encrypted websockify connections

### DISCUSSION THREAD

**Discussion by DeliDumrul43** [yume-chan/ya-webadb#474] [Q&A] [answered]
Hi to all ,

#467 I read this disccusion to have knowladge then I read **https://github.com/novnc/websockify-js** documention page to start websockify server with **wss** option and I did that now I can able to start the websockify with **wss** option but now webadb throw a error 

Executing the websockift : 
![Screenshot from 2023-01-05 14-04-50](https://user-images.githubusercontent.com/44500635/210767537-f734df59-fcd2-4602-95c4-debeb7202384.png)

After this page it gave a problem that seen below, 
![Screenshot from 2023-01-05 14-19-19](https://user-images.githubusercontent.com/44500635/210769694-f5e37ae8-69c5-4737-afe8-1030e74085d7.png)

Error , What I am doing wrong ? Please help me out.

![Screenshot from 2023-01-05 14-26-35](https://user-images.githubusercontent.com/44500635/210769861-9dcf9686-1860-4d60-b07b-8b91a5520f96.png)

**Comment by DeliDumrul43** [comment]
I got same problem with my localhost when I used **wss** option normally it would connect If I use **ws** option

**Comment by DeliDumrul43** [comment]
I inspected the web page warnings and I got these ones 
![Screenshot from 2023-01-06 10-55-54](https://user-images.githubusercontent.com/44500635/210956076-3bec67c4-793b-4ced-a51a-a8de69bbdeeb.png)
@yume-chan 

Thanks in advance

**Reply by yume-chan** [reply]
That's from Google AD, from the copycat version you are using. My website doesn't have any tracking and AD.

**Comment by yume-chan** [comment]
Make sure you entered the correct Common Name when creating the certificate using OpenSSL, that should match the IP address you are connecting (`192.168.81.25` in your case). If you created a certificate for `192.168.61.25`, it won't work for `localhsot`. Technically one certificate can validate multiple domains using "Alternative Names", but I don't know how to add them using OpenSSL, you may Google yourself.

**Reply by DeliDumrul43** [reply]
I am sure that was the correct IP address and I tried `localhost` on my PC so `wss` option didn't work then my created `.cer` file is not valid.

**Comment by yume-chan** [comment, marked as answer]
Also make sure you installed/trusted the self-issued certificate on your current PC, usually that requires you to export the certificate into `.cer` format, then open and import it.

**Reply by DeliDumrul43** [reply]
I created it I didn't install and pay it to have. So my `.cer` file is not the valid, right ?

**Reply by yume-chan** [reply]
Not paying doesn't mean the certificate is invalid. There are free certificate issuers like Let's Encrypt, however they will only issue certificates for domain names, not IP addresses. 

Your self-issued certificate is invalid because it's not from a trusted certificate issuer. Browsers trust those certificate issuers, so they also trust all certificates they issued.

But you can still trust your certificate on your own devices by adding it to the certificate store. First extract the public key using

```sh
openssl x509 -in self.pem -out self.cer
```

Then copy the `cer` file to your device running the browser, open it, and operate according to the display, or Google for your OS for detailed steps.

**Reply by DeliDumrul43** [reply]
Thanks for the information that you told I didn't know it , I am trying now
